src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php

```diff
@@ -281,13 +340,18 @@ public function deleteCartRuleAction(int $cartId, int $cartRuleId)
      */
     public function addProductAction(Request $request, int $cartId): JsonResponse
     {
-        $productId = $request->request->getInt('productId');
-        $quantity = $request->request->getInt('quantity');
-        $combinationId = $request->request->getInt('combinationId');
+        $productId = $request->request->getInt('product_id');
+        $quantity = $request->request->getInt('product_quantity');
+        $combinationId = $request->request->getInt('combination_id');
         $customizationId = null;
 
+        $textCustomizations = $request->request->get('customizations') ?: [];
+        $fileCustomizations = $request->files->get('customizations') ?: [];
+
+        $customizations = $textCustomizations + $fileCustomizations;
```

Comment:
Just a warning: `+` and `array_merge` have different behaviors (see https://stackoverflow.com/questions/5394157/whats-the-difference-between-array-merge-and-array-array) and usually `array_merge` is considered more reliable.

Reply:
array_merge doesn't work with assoc array. In this case array keys are the ids of customization fields, so i found the `+` method appropriate.

Reply:
No way!! `array_merge` doesn't manage associative arrays?! Or is it only when using integer keys?
(Seems like `array_replace` does the job https://stackoverflow.com/questions/5233721/php-array-merge-associative-arrays as an alternative Although the I prefer the`+` notation)